### PR TITLE
BHBC-1925

### PR DESCRIPTION
**Pull request by cgarrettjones**
# Overview

- adjusted permit model

## Links to Jira tickets

- [BHBC-1925](https://quartech.atlassian.net/browse/BHBC-1925)

## Description of relevant changes

- adjusted permit model for model 1.2
- adjusted db function project deletion routines
- added smoke test file

## PR Checklist

A list of items that are good to consider when making any changes.

_Note: this list is not exhaustive, and not all items are always applicable._

### Code

- [ ] New files/classes/functions have appropriately descriptive names and comment blocks to describe their use/behaviour
- [ ] I have avoided duplicating code when possible, moving re-usable pieces into functions
- [ ] I have avoided hard-coding values where possible and moved any re-usable constants to a constants file
- [ ] My code is as flat as possible (avoids deeply nested if/else blocks, promise chains, etc)
- [ ] My code changes account for null/undefined values and handle errors appropriately
- [ ] My code uses types/interfaces to help describe values/parameters/etc, help ensure type safety, and improve readability

### Style

- [ ] My code follows the established style conventions
- [ ] My code uses native material-ui components/icons/conventions when possible

### Documentation

- [ ] I have commented my code sufficiently, such that an unfamiliar developer could understand my code
- [ ] I have added/updated README's and related documentation, as needed

### Tests

- [ ] I have added/updated unit tests for any code I've added/updated
- [ ] I have added/updated the Postman requests/tests to account for any API endpoints I've added/updated

### Linting/Formatting

- [ ] I have run the linter and fixed any issues, as needed  
       _See the `lint` commands in package.json_
- [ ] I have run the formatter and fixed any issues, as needed  
       _See the `format` commands in package.json_

### SonarCloud

- [ ] I have addressed all SonarCloud Bugs, Vulnerabilities, Security Hotspots, and Code Smells

### CODE REVIEW

**Review by curtisupshall — DISMISSED** [review]
Looks good to my eyes. 👍

**Comment by AlfredRosenthal** [comment]
This all looks good. I did try running it locally just to see what happens and I think the removal of `project_id` from the `permit` table means we'll have to update a few SQL statements that are still expecting that to exist. That might have been the blow up you were seeing when you were trying to run things @cgarrettjones.

File: api/src/services/eml-service.ts [hunk 1]
```diff
@@ -409,37 +409,39 @@ export class EmlService extends DBService {
...
-            metadata: {
-              permits: {
-                permit: { permitType: item.permit.permit_type, permitNumber: item.permit.permit_number }
-              }
-            }
-          });
-        }
-      });
-    }
+    // TODO add back when survey supports permits
```
Comment (AlfredRosenthal) [inline api/src/services/eml-service.ts:412]: should these be added back now?

File: api/src/services/permit-service.test.ts [hunk 2]
```diff
@@ -1,313 +1,10 @@
...
-
-      sinon.stub(permit_queries, 'postPermitNoSamplingSQL').returns(SQL`some`);
-
-      const permitService = new PermitService(mockDBConnection);
-
-      const res = await permitService.insertNoSamplePermit(permitData, coordinatorData);
-
-      expect(res).to.equal(12);
-    });
+  it('TODO', () => {
```
Comment (AlfredRosenthal) [inline api/src/services/permit-service.test.ts]: could be removed

File: api/src/services/survey-service.ts [hunk 3]
```diff
@@ -491,31 +489,52 @@ export class SurveyService extends DBService {
...
-    await this.unassociatePermitFromSurvey(surveyId);
+  async updateSurveyPermitData(surveyId: number, surveyData: PutSurveyObject) {
+    const permitService = new PermitService(this.connection);
 
-    if (!surveyData.permit.permit_number) {
-      return;
+    // Get existing permits
+    const existingPermits = await permitService.getPermitBySurveyId(surveyId);
+
+    // Determine if any existing permits need to be deleted (are not included in the incoming permits array)
```
Comment (AlfredRosenthal) [inline api/src/services/survey-service.ts]: comment is a little confusing maybe change to
'Incoming permits array is our source of truth, find and remove any hanging/ leftover permits'

Comment (NickPhura) [reply api/src/services/survey-service.ts]: Took a stab at updating all of these comments to try and be more descriptive, not sure if its better or just different :)

File: api/src/services/permit-service.test.ts (hunk 2 above)
Comment (NickPhura) [reply api/src/services/permit-service.test.ts]: Removed, and added real tests (was just a placeholder so as not to forget)

File: api/src/services/eml-service.ts (hunk 1 above)
Comment (NickPhura) [reply api/src/services/eml-service.ts:412]: we may never add them back tbh, as its questionable if permits belong in the EML at all.  But not certain yet.

File: api/src/services/survey-service.ts (hunk 3 above)
Comment (AlfredRosenthal) [reply api/src/services/survey-service.ts]: I think it's better!

**Review by AlfredRosenthal — APPROVED** [review]
🐶

**Pull request merged**